From 78c55228a18940134a4e811ed8249fc365587cb4 Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd@luon.net>
Date: Sun, 3 Aug 2008 12:29:18 +0100
Subject: [PATCH] fix iteration over random devices

Actually iterate over the various random devices. Fixes a infinite loop on
startup when /dev/urandom isn't readable. Patch by Matt Kraai on Debian bug
491270, closes #333
---
 src/pulsecore/random.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/pulsecore/random.c b/src/pulsecore/random.c
index c72876e..ff73532 100644
--- a/src/pulsecore/random.c
+++ b/src/pulsecore/random.c
@@ -75,6 +75,8 @@ static int random_proper(void *ret_data, size_t length) {
 
         if (ret == 0)
             break;
+
+        device++;
     }
 
     return ret;
